tests/src/test/scala/akka/kafka/scaladsl/RetentionPeriodSpec.scala

```diff
@@ -48,129 +41,130 @@ class RetentionPeriodSpec extends SpecBase with TestcontainersKafkaPerClassLike
-        offset.record.partition() should equal(partition0)
-        offset.committableOffset.commitInternal()
-      }
-
-      sleep(5.seconds, "Wait for a number of offset refreshes")
-
-      // read the __consumer_offsets topic and we should see the flipping back and forth
-      val group2 = createGroupId(2)
-      val group2consumerSettings: ConsumerSettings[Array[Byte], Array[Byte]] = consumerDefaults(
-        new ByteArrayDeserializer(),
-        new ByteArrayDeserializer()
-      ).withProperty(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, "1").withGroupId(group2)
-      val probe3subscription = Subscriptions.topics("__consumer_offsets")
-      val (control3, probe3) = Consumer
-        .plainSource(group2consumerSettings.withClientId(consumerClientId3), probe3subscription)
-        .toMat(TestSink.probe)(Keep.both)
-        .run()
-      val commits: Seq[ConsumerRecord[Array[Byte], Array[Byte]]] = probe3.request(100).expectNextN(10)
-
-      // helper method ripped from GroupMetadataManager's formatter
-      def readOffset(
-          consumerRecord: ConsumerRecord[Array[Byte], Array[Byte]]
-      ): Option[(GroupTopicPartition, Option[OffsetAndMetadata])] = {
-        val offsetKey = GroupMetadataManager.readMessageKey(ByteBuffer.wrap(consumerRecord.key())) // Only read the message if it is an offset record.
-        // We ignore the timestamp of the message because GroupMetadataMessage has its own timestamp.
-        offsetKey match {
-          case offsetKey: OffsetKey =>
-            val groupTopicPartition = offsetKey.key
-            val value = consumerRecord.value
-            val formattedValue =
-              if (value == null) None else Some(GroupMetadataManager.readOffsetMessageValue(ByteBuffer.wrap(value)))
-            Some((groupTopicPartition, formattedValue))
-              .asInstanceOf[Option[(GroupTopicPartition, Option[OffsetAndMetadata])]]
-          case _: Any => None
-        }
-      }
-
-      log.debug("Ensure that commits progress in-order")
-      var progress = -1L
-      for (consumerRecord <- commits) {
-        readOffset(consumerRecord) match {
-          case Some((group, offset)) =>
-            log.debug("Committed {}: {}", group: Any, offset: Any)
-            offset match {
-              case Some(position) =>
-                if (group.topicPartition.equals(tp0)) {
-                  position.offset shouldBe >=(progress)
-                  progress = position.offset
-                }
-              case None => //noop
-            }
-          case None => // noop
-        }
-      }
-
-      // cleanup
-      probe1.cancel()
-      probe2.cancel()
-      probe3.cancel()
-
-      control1.isShutdown.futureValue shouldBe Done
-      control2.isShutdown.futureValue shouldBe Done
-      control3.isShutdown.futureValue shouldBe Done
-    }
-  }
+  // TODO: this test needs to manually consume events from __consumer_offsets without the aid of types from Kafka core
```

Comment:
Create an issue about this as well. At least we know it worked before...
Could we just copy in a few classes?

Reply:
Copying code might be a good idea.  I created an issue https://github.com/akka/alpakka-kafka/issues/1234